Sources/LanguageServerProtocol/SupportTypes/DocumentURI.swift

```diff
@@ -12,26 +12,13 @@
 
 import Foundation
 
-/// Standardize the URL.
-///
-/// This normalizes escape sequences in file URLs, like `%40` --> `@`.
-/// But for non-file URLs, has no effect.
-fileprivate func standardize(url: URL) -> URL {
-  guard url.isFileURL else { return url }
-
-  // This has the side-effect of removing trailing slashes from file URLs
-  // on Linux, so we may need to add it back.
-  let standardized = url.standardizedFileURL
-  if url.absoluteString.hasSuffix("/") && !standardized.absoluteString.hasSuffix("/") {
-    return URL(fileURLWithPath: standardized.path, isDirectory: true)
-  }
-  return standardized
-}
-
 public struct DocumentURI: Codable, Hashable {
   /// The URL that store the URIs value
   private let storage: URL
 
+  /// Standardized URL.
+  private let standardizedStorage: URL
```

Comment:
This could probably be `lazy`, which avoids the potentially expensive standardieFileURL call if you never compare the URL.

Reply:
I'll try this + `stringByRemovingPercentEncoding` to see if we can get this working without needing to do anything with the file system.